craftai/client.py

```
@@ -385,7 +587,78 @@ def _check_agent_id(agent_id):
+    valid_agent_indices = []
+    for index, agent in enumerate(payload):
+      try:
+        # Check if the agent ID is valid
+        if "id" in agent:
+          self._check_agent_id(agent["id"])
+      except CraftAiBadRequestError:
+        invalid_agent_indices.append(index)
+        invalid_agent_errors.append(CraftAiBadRequestError(ERROR_ID_MESSAGE))
+      else:
+        try:
+          # Check if the agent is serializable
+          json.dumps([agent])
+        except TypeError as e:
+          invalid_agent_indices.append(index)
+          invalid_agent_errors.append(e)
+        else:
+          valid_agent_indices.append(index)
+
+    if len(invalid_agent_indices) == len(payload):
+      raise CraftAiBadRequestError(ERROR_ID_MESSAGE)
+
+    invalid_payload = []
+    for index, invalid_agent in enumerate([payload[i] for i in invalid_agent_indices]):
+      invalid_payload.append({"id": invalid_agent["id"],
+                              "error": invalid_agent_errors[index]})
+
+    return valid_agent_indices, invalid_agent_indices, invalid_payload
+
+  @staticmethod
+  def _recreate_list_with_indices(indices1, values1, indices2, values2):
+    full_list = [None] * (len(indices1) + len(indices2))
+    for i, index in enumerate(indices1):
+      try:
+        full_list[index] = values1[i]
+      except IndexError:
+        pass
+    for i, index in enumerate(indices2):
+      try:
+        full_list[index] = values2[i]
+      except IndexError:
+        pass
+    return [x for x in full_list if x != None]
+
+  def _create_and_send_json_bulk(self, payload, req_url, request_type="GET"):
+    # Extra header in addition to the main session's
+    ct_header = {"Content-Type": "application/json; charset=utf-8"}
+
+    try:
+      json_pl = json.dumps(payload)
+    except TypeError as e:
+      raise CraftAiBadRequestError("Invalid configuration or agent id given to create JSON. {}"
+                                   .format(e.__str__()))
```

I don't understand why `json.dumps` should raise an *Invalid configuration or id* error.